//Почти сделал сепокку

Exc1usiveRUS opened the pull request:
Добавить новый функционал, HTTP сервер и обработчики запросов

Commit:
//Почти сделал сепокку

Добавить новый функционал, HTTP сервер и обработчики запросов

Commit:
//Почти сделал сепокку

Добавить новый функционал, HTTP сервер и обработчики запросов #2

Yang-Pi commented on src/manager/http/HttpTaskServer.java:
```diff
@@ -0,0 +1,56 @@
+
+    protected Gson gson;
+
+    protected TaskManager taskManager;
```
если класс не спроектирован специально для наследования от него, то не стоит помечать поля protected - по умолчанию все поля ставим private. Повышаем уровень видимости только тогда, когда понимаем, что это необходимо.
Про это еще хорошо написано в супер-книге Effective Java, item 19

Yang-Pi commented on src/manager/http/adapters/LocalDateTimeAdapter.java:
```diff
@@ -0,0 +1,32 @@
+
+public class LocalDateTimeAdapter extends TypeAdapter<LocalDateTime> {
+
+    DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm");
```
здесь тоже забыл private

Yang-Pi commented on src/manager/http/handlers/SubTaskHandlers.java:
```diff
@@ -0,0 +1,53 @@
+                case UNKNOWN ->
+                    sendNotFound(exchange);
+            }
+        } catch (NullPointerException e) {
```
в работе получилось много конструкций try-catch в каждом обработчике. В целом обработки очень похожи. И можно пойти дальше - не зря же есть BaseHttpHandler) В нем в методе handle() можно 1 раз написать блок try-catch и обработать все исключения. Внутри try {} там будет вызов некого абстрактного метода safeHandle() - который ты как раз и переопределишь в наследниках. Такой паттерн называется "Шаблонный метод" (если интересно можешь найти его на refactoring.guru). Он даст тебе возможность не дублировать код в обработчиках и написать его более компактно

Yang-Pi reviewed (commented):
хорошая работа получилась, оставил несколько рекомендаций

Commit:
//Почти сделал сепокку

Поменял области видимости и изменил BaseHttpHandler и его наследников согласно рекомендациям

Commit:
//Почти сделал сепокку

Поменял области видимости и изменил BaseHttpHandler и его наследников согласно рекомендациям #2
